### PR TITLE
Made changes to the file in order to pass the pt file

### PR DESCRIPTION
**Pull request by ajay-vishnu**
Closes the issue #109. 
I believe that there are still few changes to be made, any advise is welcome.

### CODE REVIEW

**Comment by mmirman** [comment]
@abhigya-sodani can you do a review?

**Comment by abhigya-sodani** [comment]
Hey, just tested your branch, I dont think you are passing the new model when we generate a completion using call_small, only when we refinetune call small. I.e look at like 238 of optimize.py

**Comment by ajay-vishnu** [comment]
The model path will be stored in **kwargs which is being passed when the small_model is called in the line 238 and the CALL_SMALL in line 64 of client.py first checks if the path is given and would load the model if given and then proceed to call the generate function.

**Comment by cartazio** [comment]
whats a an example test script that would demonstrate and exercise this change?

**Comment by ajay-vishnu** [comment]
So the end user can pass the parameter `small_model_filename` while calling the `complete` function like shown below:

```
# import our client
from llm_vm.client import Client
from llm_vm.config import settings
import os

# Instantiate the client specifying which LLM you want to use
client = Client(big_model='neo', small_model='pythia')

# specify the file name of the finetuned model to load
small_model_filename = '2023-08-06T13:18:12_pythia-70m-deduped.pt'
# big_model_filename = '2023-08-06T13:18:12_gpt-neo-1.3B.pt'

# Put in your prompt and go!
response = client.complete(prompt = "Answer question Q. ",context="Q: What is the currency in myanmmar",
                           openai_key=settings.openai_api_key,
                           small_model_filename=small_model_filename,
                           finetune=True)
print(response)
```

**Comment by ajay-vishnu** [comment]
But I don't see the model being updated and I'm unable to trace since the functions being called are asynchronous. And could you tell me what sort of script should I run in order to pass the `if` condition in line 236 and call the `self.call_small()` in line 238 of _optimize.py_?

**Comment by abhigya-sodani** [comment]
right now you shouldn't have to pass a model in for it to use a trained model, that should be stored in the c_id.  Line 234 gives us if a given cid already has a model assigned to it 
```
model = self.storage.get_model(c_id)
```
this is the model that we need to load for generation and finetuning. It should be done whenever a prompt with a context window that has already been finetuned for (i.e a cid that already has a model for it stored in memory) is called again. Right now this "memory" only exists within a singular runtime so the way I test it is :
```
#! /usr/bin/env python3
# import our client
from llm_vm.client import Client
import os
import time
#from llm_vm.config import settings
# Instantiate the client specifying which LLM you want to use
client = Client(big_model='chat_gpt', small_model='pythia')

# Put in your prompt and go!
response = client.complete(context= "Answer question Q. ",prompt="Q: What is the currency in myanmmar",
                           openai_key="",
                           temperature=0.0,
                           data_synthesis=True,
                           finetune=True,)
print(response)
time.sleep(60)
response = client.complete(context= "Answer question Q. ",prompt="Q: What is the economic situation in France",
                           openai_key="",
                           temperature=0.0,
                           data_synthesis=False,
                           finetune=False,)
print(response)
```

The delay is there to give the async pythia training time to complete before making the second call. Notice how the context is the same for both calls, so for the second call the trained model should be used. I think the current quickstart_finetune has the prompt and context parameters reversed right now.

**Comment by abhigya-sodani** [comment]
Tested and seems to work
@cartazio do you want to review before we push. 
Also @mmirman let's send some merch over

**Comment by cartazio** [comment]
I’ll take a close reading tomorrow / Monday

**Comment by abhigya-sodani** [comment]
@cartazio did you get a chance to review?

**Comment by cartazio** [comment]
nope, forgot, i'll look later today / tomorrow

File: src/llm_vm/completion/optimize.py [hunk 1]
```diff
@@ -264,7 +265,7 @@ def actual_train(use_completion = None):
                     if len(training_exs) >= self.MIN_TRAIN_EXS and not self.storage.get_training_in_progress_set_true(c_id):
                         print("Actually Fine-tuning", flush=True)
                         print("Training examples:",str(len(training_exs)))
-                        asyncStart(self.small_model.finetune(training_exs,self,c_id))
+                        asyncStart(self.small_model.finetune(training_exs,self,c_id,small_model_filename))
```
Comment (cartazio) [inline src/llm_vm/completion/optimize.py:268]: shouldn't this use the kwarg version of the last arg?

File: src/llm_vm/onsite_llm.py [hunk 2]
```diff
@@ -172,7 +172,7 @@ def asynctune():
                 timestamp = datetime.now().strftime('%Y-%m-%dT%H-%M-%S')
             else:
                 timestamp = datetime.now().strftime('%Y-%m-%dT%H:%M:%S')
-            new_model = os.path.join(model_path_default,"finetuned_models",self.model_name, timestamp + '_' + self.model_name + ".pt" )
+            new_model = os.path.join(model_path_default,"finetuned_models",self.model_name, timestamp + '_' + self.model_name + ".pt" ) if model_filename is None else os.path.join(model_path_default,"finetuned_models",model_filename)
```
Comment (cartazio) [inline src/llm_vm/onsite_llm.py:175]: so this has the issue that it clobbers the loaded model with the fine tuned name right
?

**Pull request merged**